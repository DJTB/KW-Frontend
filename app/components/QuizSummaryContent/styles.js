import styled from 'styled-components';

import { gutter } from 'shared/styles/layout';

export const Heading = styled.div`
  ${gutter({ type: 'outer' })}
  display: flex;
`;
