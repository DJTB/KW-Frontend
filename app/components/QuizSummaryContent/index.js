import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import H2 from 'base/H2';
import Container from 'base/Container';
import PageWrapper from 'base/PageWrapper';
import AccuracyBar from 'components/AccuracyBar';
import SummarySection from 'components/SummarySection';
import ToggleVocabListButton from 'components/ToggleVocabListButton';

import {
  selectCorrectIds,
  selectIncorrectIds,
  selectCriticalIds,
  selectPercentCorrect,
} from 'containers/App/selectors';

import { Heading } from './styles';

QuizSummaryContent.propTypes = {
  correctIds: PropTypes.array.isRequired,
  incorrectIds: PropTypes.array.isRequired,
  criticalIds: PropTypes.array.isRequired,
  percentCorrect: PropTypes.number.isRequired,
};

function QuizSummaryContent({
  correctIds,
  incorrectIds,
  criticalIds,
  percentCorrect,
}) {
  const noHistory = !incorrectIds.length && !correctIds.length && !criticalIds.length;
  return (
    <PageWrapper>
      <Heading>
        <AccuracyBar percent={percentCorrect} />
        <ToggleVocabListButton />
      </Heading>
      {noHistory ? (
        <Container><H2>No history. Get quizzing!</H2></Container>
      ) : (
        <div>
          <SummarySection
            summaryType="INCORRECT"
            ids={incorrectIds}
          />
          <SummarySection
            summaryType="CORRECT"
            ids={correctIds}
          />
          <SummarySection
            summaryType="CRITICAL"
            ids={criticalIds}
          />
        </div>
      )}
    </PageWrapper>
  );
}

const mapStateToProps = createStructuredSelector({
  correctIds: selectCorrectIds,
  incorrectIds: selectIncorrectIds,
  criticalIds: selectCriticalIds,
  percentCorrect: selectPercentCorrect,
});

export default connect(mapStateToProps)(QuizSummaryContent);
