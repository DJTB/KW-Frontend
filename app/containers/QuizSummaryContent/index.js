import React from "react";
import PropTypes from "prop-types";
import { compose, withStateHandlers } from "recompose";

import distanceInWordsToNow from "date-fns/distance_in_words_to_now";
import shouldUpdateDeepEqual from "utils/shouldUpdateDeepEqual";

import Container from "base/Container";
import H2 from "base/H2";
import H4 from "base/H4";
import PageWrapper from "base/PageWrapper";
import AccuracyBar from "components/AccuracyBar";
import SummarySection from "containers/SummarySection";
import ToggleVocabListButton from "components/ToggleVocabListButton";

import { Heading } from "./styles";

QuizSummaryContent.propTypes = {
  correctIds: PropTypes.array.isRequired,
  incorrectIds: PropTypes.array.isRequired,
  criticalIds: PropTypes.array.isRequired,
  percentCorrect: PropTypes.number.isRequired,
  cardsExpanded: PropTypes.bool.isRequired,
  toggleCardsExpanded: PropTypes.func.isRequired,
  lastActivityDate: PropTypes.oneOfType([
    PropTypes.instanceOf(Date),
    PropTypes.oneOf([false]),
  ]).isRequired,
};

function QuizSummaryContent({
  correctIds,
  incorrectIds,
  criticalIds,
  percentCorrect,
  lastActivityDate,
  cardsExpanded,
  toggleCardsExpanded,
}) {
  const noHistory = !incorrectIds.length && !correctIds.length && !criticalIds.length;
  return (
    <PageWrapper>
      <Heading>
        <AccuracyBar percent={percentCorrect} />
        <ToggleVocabListButton
          cardsExpanded={cardsExpanded}
          toggleCardsExpanded={toggleCardsExpanded}
        />
      </Heading>
      {noHistory ? (
        <Container>
          <H2>No recent history.</H2>
        </Container>
      ) : (
        <div>
          <SummarySection
            summaryType="INCORRECT"
            ids={incorrectIds}
            cardsExpanded={cardsExpanded}
          />
          <SummarySection
            summaryType="CORRECT"
            ids={correctIds}
            cardsExpanded={cardsExpanded}
          />
          {criticalIds.length > 0 && (
            <SummarySection
              summaryType="CRITICAL"
              ids={criticalIds}
              cardsExpanded={cardsExpanded}
            />
          )}
          {lastActivityDate && (
            <Container>
              <H4>
                Last session activity:{" "}
                {distanceInWordsToNow(lastActivityDate, { includeSeconds: true })} ago
              </H4>
            </Container>
          )}
        </div>
      )}
    </PageWrapper>
  );
}

export default compose(
  withStateHandlers(
    { cardsExpanded: false },
    {
      toggleCardsExpanded: ({ cardsExpanded }) => () => ({
        cardsExpanded: !cardsExpanded,
      }),
    }
  ),
  shouldUpdateDeepEqual([
    "correctIds",
    "incorrectIds",
    "criticalIds",
    "percentCorrect",
    "lastActivityDate",
    "cardsExpanded",
  ])
)(QuizSummaryContent);
