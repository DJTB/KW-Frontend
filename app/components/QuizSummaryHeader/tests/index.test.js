import React from 'react';
import { shallow } from 'enzyme';
import QuizSummaryContent from '../index';

describe('<QuizSummaryContent />', () => {
  it('should match baseline snapshot', () => {
    const renderedComponent = shallow(
      <QuizSummaryContent
        category="review"
        linkRoute="/reviews"
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should match adopt count prop', () => {
    const renderedComponent = shallow(
      <QuizSummaryContent
        category="lesson"
        linkRoute="/lessons"
        count={22}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
