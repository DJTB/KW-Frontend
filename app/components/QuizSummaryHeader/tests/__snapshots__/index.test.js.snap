// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizSummaryContent /> should match adopt count prop 1`] = `
<styled.header>
  <styled.nav>
    <LogoLink
      size="4.5rem"
      to="/"
    />
    <Styled(styled.h1)>
      Lesson
       Summary
    </Styled(styled.h1)>
    <SessionLink
      color="blue"
      count={22}
      text="Continue Session"
      to="/lessons/"
    />
  </styled.nav>
</styled.header>
`;

exports[`<QuizSummaryContent /> should match baseline snapshot 1`] = `
<styled.header>
  <styled.nav>
    <LogoLink
      size="4.5rem"
      to="/"
    />
    <Styled(styled.h1)>
      Review
       Summary
    </Styled(styled.h1)>
    <SessionLink
      color="blue"
      count={0}
      text="Continue Session"
      to="/reviews/"
    />
  </styled.nav>
</styled.header>
`;
